out/extension.js

```diff
@@ -207,8 +215,23 @@ function activate(context) {
     ], new splunkFoldingRangeProvider.confFoldingRangeProvider()));
 
     // If vscode was opened with an active Splunk file, handle it.
-    if(vscode.window.activeTextEditor && isSplunkFile(vscode.window.activeTextEditor.document.fileName)) {
-        handleSplunkFile(context);
+    vscode.commands.registerCommand('splunk.restartSpl2LanguageServer', async () => {
+        try {
+            if (spl2Client) {
+                await spl2Client.deactivate();
+            }
+            spl2Client = undefined;
+            await handleSpl2File(context, progressBar);
+        } catch (err) {
+            console.warn(`Error restarting SPL2 language server, err: ${err}`);
+        }
+    });
+    if(vscode.window.activeTextEditor) {
+        if (isSplunkFile(vscode.window.activeTextEditor.document.fileName)) {
+            handleSplunkFile(context);
+        } else if (vscode.window.activeTextEditor.document.languageId == 'splunk_spl2') {
```

Comment:
What's the logic here? Why is line 230 have a more generic condition? isn't supposed to also check for languageId or check for extension? such as `=== ".spl"`

Reply:
good call I'll move this logic to a helper function per your comment below